cmake_modules/commonSetup.cmake

```diff
@@ -58,6 +58,7 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
   option(USE_OPENLDAP "Enable OpenLDAP support (requires OpenLDAP)" ON)
   option(USE_ICU "Enable unicode support (requires ICU)" ON)
   option(USE_BOOST_REGEX "Configure use of boost regex" ON)
+  option(CENTOS_6_BOOST "Supply regex library on CentOS 6" OFF)
```

Comment:
Is there a reason we can't detect when this needs to be set?

Reply:
I'll port over some logic to detect os and version so we can start automating this kind of thing instead of expecting the developer to set the flags.  https://track.hpccsystems.com/browse/HPCC-22859